backend/app/api/v1/repo_stats.py

```
@@ -0,0 +1,185 @@
+    ]
+
+    for pattern, _ in patterns:
+        match = re.search(pattern, repo_input)
+        if match:
+            owner, repo = match.groups()
+            return owner, repo
+
+    raise ValueError(
+        f"Invalid repository format: '{repo_input}'. "
+        "Expected: 'owner/repo' or 'https://github.com/owner/repo'"
+    )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)